components/esm/esmreader.cpp

```diff
@@ -59,6 +59,14 @@ void ESMReader::clearCtx()
 
 void ESMReader::resolveParentFileIndices(const std::vector<ESMReader>& allPlugins)
 {
+    mCtx.index = -1;
+    for (size_t i=0; i<allPlugins.size(); ++i)
```

Comment:
This is `std::find_if` basically:
```c++
const auto it = std::find_if(allPlugins.begin(), allPlugins.end(), [&] (const auto& v) { return &v == this; });
if (it == allPlugins.end())
    throw std::runtime_error("Matching to ESMReader plugin is not found");
mCtx.index = it - allPlugins.begin();
```

But also this code doesn't look good. Why do you need to find the object by a pointer? Why can't you set the index when `allPlugins` is created. It should be constant complexity and now it's linear.

Reply:
I did not spend too much time here on prettification and optimisation because this PR removes a blocking issue.

Reply:
> Why can't you set the index when allPlugins is created.

Explicit and redundant setting creates a potential for errors.

Reply:
> It should be constant complexity and now it's linear.

I do not see any indications that the performance of this particular code ever matters.

Reply:
> Explicit and redundant setting creates a potential for errors.

Explicit and redundant lookups creates a potential for errors. All you need is to set an index value to `mCtx.index`. It can be done when element is pushed to the vector. There is no need to store additional state.

> I do not see any indications that the performance of this particular code ever matters.

This is not only about performance. It's harder to follow the data flow. What are guarantees that `this` should be present in `allPlugins`?

Reply:
I suppose it is a bit subjective from which side we want to verify. It depends on which trip ups we want to prevent the most. Suggestions made will be applied now,  
but I do feel they might be a bit nitpicky considering the urgency of the situation @uramer and @petrmikheev had implied.